src/components/List/List.js

```diff
@@ -0,0 +1,110 @@
+        this.editFill = this.editFill.bind(this);
+        this.editSave = this.editSave.bind(this);
+    }
+
+    sortByName() {
+        this.state.data.sort( function ( a, b ) {
+            if ( a.values_.street < b.values_.street ){
+              return -1;
+            }
+            if ( a.last_nom > b.last_nom ){
+              return 1;
+            }
+            return 0;
+          });
+          this.setState({unsortedData: this.state.data})
+    }
+
+    deleteElement(id) {
+        document.getElementById('edit-box').style.display = "none";
+        document.getElementById('i-search-street').value = "";
+        vectorLayer.getSource().removeFeature(vectorLayer.getSource().uidIndex_[id]);
+        this.setState({data: vectorLayer.getSource().getFeatures(), unsortedData: vectorLayer.getSource().getFeatures()});
+    }
+
+    filterElements(val) {
+        let leftPlaces = this.state.unsortedData.filter(function (e) {
+            return e.values_.street.substring(0,val.length).toLowerCase() == val.toLowerCase();
+        });
+        this.setState({data: leftPlaces});
+    }
+
+    editFill(id) {
+        document.getElementById('edit-box').style.display = "block";
+        document.getElementById('p-street').textContent = vectorLayer.getSource().uidIndex_[id].values_.street;
+        document.getElementById('i-spots').value = vectorLayer.getSource().uidIndex_[id].values_.spots;
+        document.getElementById('i-handicappedSpots').value = vectorLayer.getSource().uidIndex_[id].values_.handicappedSpots;
+        document.getElementById('i-paid').checked = vectorLayer.getSource().uidIndex_[id].values_.paid;
+        document.getElementById('i-id').value = id;
+        
+    }
+
+    editSave(id) {
+        vectorLayer.getSource().uidIndex_[id].values_.spots = document.getElementById('i-spots').value;
+        vectorLayer.getSource().uidIndex_[id].values_.handicappedSpots = document.getElementById('i-handicappedSpots').value;
+        vectorLayer.getSource().uidIndex_[id].values_.paid = document.getElementById('i-paid').checked;
+        this.setState({data: vectorLayer.getSource().getFeatures(), unsortedData: vectorLayer.getSource().getFeatures()});
+        document.getElementById('edit-box').style.display = "none";
+    }
+    
+    render() {
+        return (
+            <div>
+                <table cellSpacing="0">
+                    <thead>
+                        <tr>
+                            <th className="btn-sort" onClick={this.sortByName}>Ulica</th>
+                            <th>Miejsca</th>
+                            <th>Dla niep.</th>
+                            <th>Płatny</th>
+                            <th></th>
+                            <th></th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {this.state.data.map(item => (
+                            <ListItem 
+                                street={item.values_.street} 
```

Comment:
zamiast korzystać z prywatnego pola values_, można skorzystać z propertiesów obiektu OL

jest to obiekt, w którym możemy "przechowywać" dowolne wartości, które będą nam potrzebne w interakcji z obiektem

https://openlayers.org/en/latest/apidoc/module-ol_Feature-Feature.html#setProperties
https://openlayers.org/en/latest/apidoc/module-ol_Feature-Feature.html#getProperties